config/pod.cache2disk.yaml.tim

```diff
@@ -0,0 +1,105 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: "{{pod_name}}"
+  labels:
+    sdbuild: "{{build_id_with_prefix}}"
+    app: screwdriver-vm
+    tier: builds
+spec:
+  affinity:
+    podAntiAffinity:
+      preferredDuringSchedulingIgnoredDuringExecution:
+      - weight: 100
+        podAffinityTerm:
+          topologyKey: kubernetes.io/hostname
+          labelSelector:
+            matchExpressions:
+            - key: app
+              operator: In
+              values:
+              - screwdriver-vm
+  restartPolicy: Never
+  containers:
+  - name: vm-launcher
+    image: {{base_image}}
+    imagePullPolicy: Always
+    lifecycle:
+      preStop:
+        exec:
+          command: ["/bin/sh","-c","sleep 30"]
+    securityContext:
+      privileged: true
+    resources:
+      limits:
+        cpu: "{{cpu}}"
+        memory: {{memory}}Mi
+    command: ["/sd/hyper-runner.sh"]
+    args: [
+         "--cpu", "{{cpu}}",
+         "--memory", "{{memory}}",
+         "--container", "{{container}}",
+         "--api_uri", "{{api_uri}}",
+         "--store_uri", "{{store_uri}}",
+         "--ui_uri", "{{ui_uri}}",
+         "--build_id", "{{build_id}}",
+         "--job_id", "{{job_id}}",
+         "--event_id","{{event_id}}",
+         "--pipeline_id", "{{pipeline_id}}",
+         "--id_with_prefix", "{{build_id_with_prefix}}",
+         "--build_token", "{{token}}",
+         "--build_timeout", "{{build_timeout}}",
+         "--launcher_version", "{{launcher_version}}",
+         "--cache_strategy", "{{cache_strategy}}",
+         "--cache_path", "{{cache_path}}"
+          ]
+    env:
+    - name: PUSHGATEWAY_URL
+      value: "{{pushgateway_url}}"
+    - name: NODE_ID
+      valueFrom:
+        fieldRef:
+          fieldPath: spec.nodeName
+    volumeMounts:
+    - mountPath: /var/run
+      name: hyper-socket
+    - mountPath: /var/sd-workspaces
+      name: sd-workspaces
+    - mountPath: /opt/sd
+      name: sdlauncher
+    - mountPath: /opt/sd/cache/pipeline
+      name: sd-pipeline-cache
+    - mountPath: /opt/sd/cache/job
+      name: sd-job-cache
+    - mountPath: /opt/sd/cache/event
+      name: sd-event-cache
+  initContainers:
+  - name: launcher
+    image: {{launcher_image}}
+    command: ['/bin/sh', '-c', 'if ! [ -f /opt/launcher/launch ]; then TEMP_DIR=`mktemp -d -p /opt/launcher` && cp -a /opt/sd/* $TEMP_DIR && mkdir -p $TEMP_DIR/hab && cp -a /hab/* $TEMP_DIR/hab && mv $TEMP_DIR/* /opt/launcher && rm -rf $TEMP_DIR || true; else ls /opt/launcher; fi;']
+    volumeMounts:
+    - mountPath: /opt/launcher
+      name: sdlauncher
+  volumes:
+    - name: hyper-socket
+      hostPath:
+        path: /var/run
+    - name: sd-workspaces
+      hostPath:
+        path: /opt/screwdriver
+    - name: sdlauncher
+      type: DirectoryOrCreate
+      hostPath:
+        path: /opt/screwdriver/sdlauncher/{{launcher_version}}
+    - name: sd-pipeline-cache
```

Comment:
is there a way that we can look at the top level `cache` configuration on the screwdriver.yaml to create the necessary volume?

Reply:
I don't think so

Reply:
The reason I ask is that the `cache-bookend` knows about that the cache levels

Reply:
cache-bookend knows cache levels from screwdriver.yaml? why u want to restrict only specific volume? why not all pipeline, event n job? actually we don't need 3 volumes, in the long run, all can be combined into a single volume. As of now, I'm keeping 3 to honor the cache scope

Reply:
One value I see in moving this logic to cache-bookend or store-cli  is to make the dir creations on-demand. With current approach, we will have tons of empty directories whether they are used or not. 
So we mount only the cache path and rest handled by clients like cache-bookend or store-cli.

But argument against this is will be on how do we enforce scopes. If clients can create or remove paths, then any build step can do it.
Also for Pull request builds we need to make pipeline cache read-only.

Reply:
Generating specific scoped folders here will ensure that clients can't treat it like a free flow directory and we can enforce read/write protections for PR builds.